### PR TITLE
Add config option to disable registration

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

**Comment by erikjohnston** [comment]
Sigh, and now `LegacyRegister` is above the gocyclo limit

**Comment by richvdh** [comment]
> Sigh, and now LegacyRegister is above the gocyclo limit

It doesn't fit on my screen, and tbh makes my eyes hurt. I think splitting it up would be a good thing.

File: src/github.com/matrix-org/dendrite/common/config/config.go [hunk 1]
```diff
@@ -83,6 +83,9 @@ type Dendrite struct {
 		// If set, allows registration by anyone who also has the shared
 		// secret, even if registration is otherwise disabled.
 		RegistrationSharedSecret string `yaml:"registration_shared_secret"`
+		// If set disables new users from registering (excpet via shared
```
Comment (richvdh) [inline src/github.com/matrix-org/dendrite/common/config/config.go]: excpet

**Comment by erikjohnston** [comment]
I've split out the parsing and validation into a separate function, which seemed the sanest way of splitting it.

**Pull request merged**